src/lib.rs

```diff
@@ -112,12 +112,29 @@ use std::collections::HashSet;
 use std::path;
 use walrus::ir::VisitorMut;
 
+/// Input configuration.
+#[derive(Clone, Debug)]
+pub enum Input {
+    /// The input `.wasm` file that should have its function snipped.
+    File(path::PathBuf),
+    /// The input WebAssembly blob that should have its function snipped.
+    Buffer(Vec<u8>),
+    // TODO: Support Walrus module directly.
+    // Module(walrus::Module),
```

Comment:
`walrus::Module` doesn't support the `Clone` trait which is a requirement of `Options`. Would it make sense adding that to walrus?

Reply:
Unfortunately, it doesn't really make sense to support `Clone` for `walrus::Module`, since its internal arenas and the various ids don't really make sense in a `Clone` world.

I think the best way to support passing a walrus::Module directly would be to move this enum into the type level and have a type parameter that implements a fancy version of `TryInto<walrus::Module>`, and then let the `derive(Clone)` just conditionally implement `Clone` for `Options` when the type parameter is also `Clone`. This doesn't need to happen in this PR, and can instead be a follow up PR, if you're interested in implementing this strategy.

Reply:
I'm not sure I fully follow, but after integrating this I came to the realisation that it would be better not having `Input` coupled to `Options`. I propose that `snip` takes two arguments, options and input. Additionally it would solve this problem too.

Reply:
Well, that doesn't play well with the `skip_producers_section`, which is an options for walrus.

Reply:
@fitzgen I won't have time anytime soon to look into this, but could make a release of wasm-snip after this was merged?

Reply:
Done!

Reply:
Thanks @fitzgen!